OpenRA.Game/FileSystem/FileSystem.cs

```diff
@@ -164,58 +182,67 @@ public Stream Open(string filename)
 			return s;
 		}
 
-		public bool TryOpen(string name, out Stream s)
+		public bool TryGetPackageContaining(string path, out IReadOnlyPackage package, out string filename)
 		{
-			var filename = name;
-			var packageName = string.Empty;
-
-			// Used for faction specific packages; rule out false positive on Windows C:\ drive notation
-			var explicitPackage = name.Contains(':') && !Directory.Exists(Path.GetDirectoryName(name));
-			if (explicitPackage)
+			var explicitSplit = path.IndexOf('|');
+			if (explicitSplit > 0)
 			{
-				var divide = name.Split(':');
-				packageName = divide.First();
-				filename = divide.Last();
+				if (explicitMounts.TryGetValue(path.Substring(0, explicitSplit), out package))
```

Comment:
You could combine those two checks: `if (explicitSplit > 0 && explicitMounts...)`